packages/sdk/src/chains/EVM/__test__/approvesAndChecksFns.test.ts

```diff
@@ -0,0 +1,113 @@
+import { ERC721MinterBurnerPauser, ERC20 } from '@buildwithsygma/sygma-contracts';
+import { ContractReceipt, ethers, BigNumber } from 'ethers';
+import { getApproved, checkCurrentAllowanceOfErc20, approve } from '../utils/approvesAndChecksFns';
+
+describe('getApproved Function Tests', () => {
+  it('should determine whether the specified token is approved for the provided handler address', async () => {
+    const tokenId = 1;
+    const tokenInstance = {
+      getApproved: jest.fn().mockResolvedValue('0x1234567890123456789012345678901234567890'),
+    } as unknown as ERC721MinterBurnerPauser;
+    const handlerAddress = '0x1234567890123456789012345678901234567890';
+
+    const isApproved = await getApproved(tokenId, tokenInstance, handlerAddress);
+    expect(isApproved).toBeTruthy();
+  });
+
+  it('should thrown error', async () => {
+    const tokenId = 1;
+    const tokenInstance = {
+      getApproved: jest.fn().mockRejectedValue(new Error('Sick error')),
+    } as unknown as ERC721MinterBurnerPauser;
+    const handlerAddress = '0x1234567890123456789012345678901234567890';
+
+    await expect(getApproved(tokenId, tokenInstance, handlerAddress)).rejects.toThrowError(
+      'Sick error',
+    );
+  });
+});
+
+describe('checkCurrentAllowanceOfErc20', () => {
+  let senderAddress: string;
+  let erc20Instance: ERC20;
+  let erc20HandlerAddress: string;
+
+  beforeEach(() => {
+    senderAddress = '0xabc';
+    erc20Instance = {} as unknown as ERC20;
+    erc20HandlerAddress = '0x123';
+  });
+
+  it('should return the correct allowance when allowance is greater than zero', async () => {
+    const mockAllowance = { _hex: '0x10000' };
+    erc20Instance = {
+      allowance: jest.fn().mockResolvedValue(mockAllowance),
+    } as unknown as ERC20;
+
+    const result = await checkCurrentAllowanceOfErc20(
+      senderAddress,
+      erc20Instance,
+      erc20HandlerAddress,
+    );
+    expect(result.toString()).toEqual('6.5536e-14');
+  });
+
+  it('should return zero when allowance is zero', async () => {
+    const mockAllowance = { _hex: '0x0' };
+    erc20Instance = {
+      allowance: jest.fn().mockResolvedValue(mockAllowance),
+    } as unknown as ERC20;
+
+    const result = await checkCurrentAllowanceOfErc20(
+      senderAddress,
+      erc20Instance,
+      erc20HandlerAddress,
+    );
+    expect(result).toEqual(0);
+  });
+
+  it('should throw an error on failure', async () => {
+    // set up mock for a failed allowance query
```

Comment:
Thanks for providing a comment but seems it is not necessary for this situation as is obvious from test name